src/rgw/driver/d4n/d4n_policy.h

```diff
@@ -1,104 +1,138 @@
+      dir = new BlockDirectory{io};
+    }
+
+    virtual int init(CephContext *cct, const DoutPrefixProvider* dpp) {
+      std::string address = cct->_conf->rgw_local_cache_address;
+
+      config cfg;
+      cfg.addr.host = address.substr(0, address.find(":"));
+      cfg.addr.port = address.substr(address.find(":") + 1, address.length());
+
+      if (!cfg.addr.host.length() || !cfg.addr.port.length()) {
+	ldpp_dout(dpp, 10) << "RGW Redis Cache: Redis cache endpoint was not configured correctly" << dendl;
+	return -EDESTADDRREQ;
+      }
+
+      dir->init(cct, dpp);
+      conn->async_run(cfg, {}, net::consign(net::detached, conn));
+
+      return 0;
+    }
+    virtual int exist_key(std::string key, optional_yield y) override;
+    virtual int get_block(const DoutPrefixProvider* dpp, CacheBlock* block, rgw::cache::CacheDriver* cacheNode, optional_yield y) override;
+    virtual uint64_t eviction(const DoutPrefixProvider* dpp, rgw::cache::CacheDriver* cacheNode, optional_yield y) override;
+    virtual void insert(const DoutPrefixProvider* dpp, std::string& key, uint64_t offset, uint64_t len, std::string version, rgw::cache::CacheDriver* cacheNode, optional_yield y) override;
+    virtual bool erase(const DoutPrefixProvider* dpp, const std::string& key);
+    virtual void shutdown() override;
 };
 
 class LRUPolicy : public CachePolicy {
-  public:
-  struct Entry : public boost::intrusive::list_base_hook<> {
-      std::string key;
-      uint64_t offset;
-      uint64_t len;
-      Entry(std::string& key, uint64_t offset, uint64_t len) : key(key), offset(offset), len(len) {}
-  };
-  LRUPolicy() = default;
   private:
-    //The disposer object function
-    struct Entry_delete_disposer {
-      void operator()(Entry *e) {
-        delete e;
-      }
-    };
-    typedef boost::intrusive::list<Entry> List;
     List entries_lru_list;
-    std::unordered_map<std::string, Entry*> entries_map;
+
   public:
-    virtual int find_client(const DoutPrefixProvider* dpp, cpp_redis::client* client) override { return 0; };
-    virtual int exist_key(std::string key) override;
-    virtual int get_block(const DoutPrefixProvider* dpp, CacheBlock* block, rgw::cache::CacheDriver* cacheNode) override;
-    virtual uint64_t eviction(const DoutPrefixProvider* dpp, rgw::cache::CacheDriver* cacheNode) override;
-    virtual void insert(const DoutPrefixProvider* dpp, std::string& key, uint64_t offset, uint64_t len, rgw::cache::CacheDriver* cacheNode) override;
-    bool erase(const DoutPrefixProvider* dpp, const std::string& key);
+    LRUPolicy() = default;
+
+    virtual int exist_key(std::string key, optional_yield y) override;
+    virtual int get_block(const DoutPrefixProvider* dpp, CacheBlock* block, rgw::cache::CacheDriver* cacheNode, optional_yield y) override;
+    virtual uint64_t eviction(const DoutPrefixProvider* dpp, rgw::cache::CacheDriver* cacheNode, optional_yield y) override;
```

Comment:
Is eviction() always going to be called from get_block()? Then you need not expose this as an API